src/workflows/client/client.py

```diff
@@ -0,0 +1,267 @@
+        request_body = {
+            "start_event": start_event or {},
+            "context": context or {},
+            "additional_kwargs": kwargs,
+        }
+        async with self._get_client() as client:
+            response = await client.post(
+                f"/workflows/{workflow_name}/run-nowait", json=request_body
+            )
+
+            response.raise_for_status()
+
+            return response.json()
+
+    async def get_workflow_events(
+        self,
+        handler_id: str,
+    ) -> AsyncGenerator[dict[str, Any], None]:
+        """
+        Stream events as they are produced by the workflow.
+
+        Args:
+            handler_id (str): ID of the handler running the workflow
+
+        Returns:
+            AsyncGenerator[dict[str, Any], None]: Generator for the events that are streamed in the form of dictionaries.
+        """
+        url = f"/events/{handler_id}?sse=false"
+
+        async with self._get_client() as client:
+            try:
+                async with client.stream("GET", url) as response:
+                    # Handle different response codes
+                    if response.status_code == 404:
+                        raise ValueError("Handler not found")
+                    elif response.status_code == 204:
+                        # Handler completed, no more events
+                        return
+
+                    response.raise_for_status()
+
+                    async for line in response.aiter_lines():
+                        if line.strip():  # Skip empty lines
+                            try:
+                                event = json.loads(line.replace("\n", ""))
+                                yield event
+                            except json.JSONDecodeError as e:
+                                print(f"Failed to parse JSON: {e}, data: {line}")
+                                continue
+
+            except httpx.TimeoutException:
+                raise TimeoutError(
+                    f"Timeout waiting for events from handler {handler_id}"
+                )
+            except httpx.RequestError as e:
+                raise ConnectionError(f"Failed to connect to event stream: {e}")
+
+    async def send_event(
+        self,
+        handler_id: str,
+        event: Event | dict[str, Any] | str,
+        step: str | None = None,
+    ) -> bool:
+        """
+        Send an event to the workflow.
+
+        Args:
+            handler_id (str): ID of the handler of the running workflow to send the event to
+            event (Event | dict[str, Any] | str): Event to send, represented as an Event object, a dictionary or a serialized string.
+            step (str | None): Step to send the event to (optional, defaults to None)
+
+        Returns:
+            bool: Success status of the send operation
+        """
+        if isinstance(event, Event):
+            try:
+                event = event.model_dump_json()
```

Comment:
this is the wrong way to serialize. These need to be the `JsonSerializer` serialized values, otherwise the server will not know what type to deserialize to